include/onnxruntime/interface/provider/provider.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <functional>
+
+#include "interface/graph/graph.h"
+#include "interface/framework/kernel.h"
+//#include "core/session/onnxruntime_c_api.h"
+#include "core/framework/ortdevice.h"
+#include "core/framework/stream_handles.h"
+#include "core/framework/node_compute_info.h"
```

Comment:
let's carefully review the included headers. if those are the headers we are going to include (with header-only implementation), let's think about how to structure the include folders to represent the depenency.